### PR TITLE
feat(eslint-plugin-template): add rule eqeqeq

### PR DESCRIPTION
**Pull request by rafaelss95**
Relates to #202, #315 and #399.

~Waiting for #445 to mark it as ready for review.~

### CODE REVIEW

File: packages/eslint-plugin-template/src/rules/eqeqeq.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      description: 'Requires `===` and `!==` in place of `==` and `!=`',
+      category: 'Best Practices',
+      recommended: false,
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allowNilCheck: {
```
Comment (JamesHenry) [inline packages/eslint-plugin-template/src/rules/eqeqeq.ts:28]: It's confusing to use the term "nil" in the context of a language which doesn't have it IMO. Please rename the option

**Review by JamesHenry — CHANGES_REQUESTED** [review]

File: packages/eslint-plugin-template/src/rules/eqeqeq.ts (hunk 1 above)
Comment (rafaelss95) [reply packages/eslint-plugin-template/src/rules/eqeqeq.ts:28]: Hmmm, what about: 

- `allowNullish` (ref: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/sort-type-union-intersection-members.md#options);
- `allowNullOrUndefined`;
- enter a suggestion.

Comment (JamesHenry) [reply packages/eslint-plugin-template/src/rules/eqeqeq.ts:28]: Either of those is good I think, "nullish" is a term in the language now thanks to the new operator, and null or undefined is descriptive :)

Comment (JamesHenry) [reply packages/eslint-plugin-template/src/rules/eqeqeq.ts:28]: Sorry missed your final point, I'm not sure to be honest, I don't have a great sense of which is more "idiomatic" for equivalent options in equivalent rules

**Pull request merged**